gradoop-examples/src/main/java/org/gradoop/benchmark/sampling/SamplingBenchmark.java

```diff
@@ -0,0 +1,421 @@
+
+  /**
+   * Option to to set whether graph statistics are to be computed
+   */
+  private static final String OPTION_COMPUTE_STATISTICS = "s";
+
+  /**
+   * Used input path
+   */
+  private static String INPUT_PATH;
+
+  /**
+   * Used output path
+   */
+  private static String OUTPUT_PATH;
+
+  /**
+   * Output path default
+   */
+  private static final String OUTPUT_PATH_DEFAULT = "./sampling_benchmark/";
+
+  /**
+   * Output path suffix defining where resulting graph sample is written to
+   */
+  private static final String OUTPUT_PATH_GRAPH_SAMPLE_SUFFIX = "graph_sample/";
+
+  /**
+   * Output path suffix defining where resulting benchmark file is written to
+   */
+  private static final String OUTPUT_PATH_BENCHMARK_SUFFIX = "benchmark";
+
+  /**
+   * Output path suffix defining where input graph statistics are written to
+   */
+  private static final String GRAPH_STATISTICS_ORIGIN = "graph_statistics/origin/";
+
+  /**
+   * Output path suffix defining where sample graph statistics are written to
+   */
+  private static final String GRAPH_STATISTICS_SAMPLE = "graph_statistics/sample/";
+
+  /**
+   * Integer defining the sampling algorithm that is to be evaluated.
+   */
+  private static int SELECTED_ALGORITHM;
+
+  /**
+   * List of parameters that are used to instantiate the selected sampling algorithm.
+   */
+  private static String[] CONSTRUCTOR_PARAMS;
+
+  /**
+   * Boolean determining if graph statistics for input and sample graph shall be computed
+   */
+  private static boolean COMPUTE_STATISTICS;
+
+  /**
+   * Enum of available sampling algorithms. Enum is constructed with meta information about a given
+   * sampling algorithm like its name.
+   */
+  private enum Algorithm {
+    /** PageRankSampling enum constant */
+    PAGE_RANK_SAMPLING(PageRankSampling.class.getSimpleName()),
+    /** RandomEdgeSampling enum constant */
+    RANDOM_EDGE_SAMPLING(RandomEdgeSampling.class.getSimpleName()),
+    /** RandomLimitedDegreeVertexSampling enum constant */
+    RANDOM_LIMITED_DEGREE_VERTEX_SAMPLING(RandomLimitedDegreeVertexSampling.class.getSimpleName()),
+    /** RandomNonUniformVertexSampling enum constant */
+    RANDOM_NON_UNIFORM_VERTEX_SAMPLING(RandomNonUniformVertexSampling.class.getSimpleName()),
+    /** RandomVertexEdgeSampling enum constant */
+    RANDOM_VERTEX_EDGE_SAMPLING(RandomVertexEdgeSampling.class.getSimpleName()),
+    /** RandomVertexNeighborhoodSampling enum constant */
+    RANDOM_VERTEX_NEIGHBORHOOD_SAMPLING(RandomVertexNeighborhoodSampling.class.getSimpleName()),
+    /** RandomVertexSampling enum constant */
+    RANDOM_VERTEX_SAMPLING(RandomVertexSampling.class.getSimpleName());
+
+    /** Property denoting the simple classname of a sampling algorithm */
+    private final String name;
+
+    /**
+     * Enum constructor
+     * @param name simplified class name
+     */
+    Algorithm(String name) {
+      this.name = name;
+    }
+  }
+
+  static {
```

Comment:
keep the code style persistent (e.g. GroupingBenchmark)

Reply:
could you please describe in more detail what you're criticizing here?
do you mean the distance of the static block to the enum block?

Reply:
You mixed up the option construction there:

```
OPTIONS.addOption(Option.builder(OPTION_CONSTRUCTOR_PARAMS)
      .longOpt("params")
      .hasArgs()
      .desc("Whitespace separated list of algorithm parameters")
      .required()
      .build());
    OPTIONS.addOption(OPTION_OUTPUT_PATH, "output", true,
      "Path to directory where resulting graph sample, benchmark file and graph " +
        "statistics are written to. (Defaults to " + OUTPUT_PATH_DEFAULT + ")");
```
I prefer to keep it consistent with other benchmark runners. That would be without `Option.builder`.

Reply:
I chose to use the `Option.builder` in some cases because it allows for the construction of required options (which is useful because then there is no need for manual checks). I just checked the api doc again and apparently there is also a `addRequiredOption` method which i will use instead.